### PR TITLE
feat(expo): update to SDK 52

### PR DESCRIPTION
**Pull request by ecreeth**
This PR is part of https://github.com/facebook/react-strict-dom/issues/236 [action items](Update to SDK 52)

#### Test Plan:
Screenshot 2024-11-12 at 8 05 38 PM
Screenshot 2024-11-12 at 8 02 29 PM
#### IOS & Android

Screenshot 2024-11-12 at 7 56 35 PM

### CODE REVIEW

File: apps/examples/src/index.js [hunk 1]
```diff
@@ -7,9 +7,7 @@
 
 import { LogBox } from 'react-native';
 
-// Taken from "npm:expo/AppEntry.js" to fix broken "package.main"
-// in Expo template when used in npm workspaces
-import registerRootComponent from 'expo/build/launch/registerRootComponent';
+import { registerRootComponent } from 'expo';
```
Comment (ecreeth) [inline apps/examples/src/index.js:10]: Based on https://docs.expo.dev/guides/monorepos/#change-default-entrypoint

File: apps/examples/app.json [hunk 2]
```diff
@@ -6,14 +6,13 @@
     "orientation": "portrait",
     "icon": "./assets/icon.png",
     "userInterfaceStyle": "light",
+    "newArchEnabled": true,
```
Comment (ecreeth) [inline apps/examples/app.json:9]: https://docs.expo.dev/guides/new-architecture/#enable-the-new-architecture-in-an-existing-project

File: apps/examples/src/index.js (hunk 1 above)
Comment (ecreeth) [reply apps/examples/src/index.js:10]: This fixes:
Screenshot 2024-11-12 at 8 03 45 PM

Comment (ecreeth) [reply apps/examples/src/index.js:10]: This also fixes the error: 
```js
import registerRootComponent from 'expo/src/launch/registerRootComponent';
```

**Comment by ecreeth** [comment]
I verified that the `mark` element issue in `ios` still exists:
Screenshot 2024-11-12 at 8 44 14 PM

**Comment by necolas** [comment]
Thanks, looks good. Can confirm this has turned on features previously missing, like support for `position:static`.

**Pull request closed without merging**